test/files/rules/noinferrabletypes.test.ts

```diff
@@ -0,0 +1,21 @@
+// errors, inferrable type is declared
+let x: number = 7;
+let y: boolean = false;
+let z: string = "foo";
+
+// errors, types are inferrable
+function foo (a: number = 5, b: boolean = true, c: string = "bah") { }
+
+// not errors, inferrable type is not declared
+let _x = 7;
+let _y = false;
+let _z = "foo";
+
+// not error, type is not inferrable
+let weird: any = 123;
+
+// not errors, inferrable type is not declared
+function bar(a = 5, b = true, c = "bah") { }
+
+// not errors, types are not inferrable
+function baz(a: any = 5, b: any = true, c: any = "bah") { }
```

Comment:
this might be out of scope for this PR, but I wonder if we can lint for inferrable return types from functions / class methods?

for example, if a function doesn't `return`, we infer its return type as `void`. we use this convention in our rule walkers (at one point I did a pass and removed a lot of `: void` annotations).